### DISCUSSION TITLE
Avoiding XXE attacks

### DISCUSSION THREAD

**Discussion by applyinnovations** [NaturalIntelligence/fast-xml-parser#353] [Q&A]
Hey

We are considering using fast-xml-parser to receive and parse xml in our production system. Does this library provide any ability to disable DOCTYPE definitions or external entities? We would like to secure our endpoint against XXE attacks in case one of our API keys was leaked.

Thanks,
Alex

**Comment by applyinnovations** [comment]
Is anyone able to answer this or make a recommendation?

**Reply by pluma** [reply]
It doesn't look like this library respects DOCTYPE definitions at all. I did a quick test and it seems entities are passed back as-is and doctypes are stripped.

**Reply by applyinnovations** [reply]
Thanks pluma. I ended up coming to the same conclusion

**Comment by amitguptagwl** [comment]
DOCTYPE entities are supported from v4 with a few limitations to avoid attacks. Please read [Entities](https://github.com/NaturalIntelligence/fast-xml-parser/blob/master/docs/5.Entities.md) for more detail.
